client/login/controller.js

```diff
@@ -135,6 +136,19 @@ export async function login( context, next ) {
 	next();
 }
 
+export async function desktopLogin( context, next ) {
+	context.primary = <DesktopLogin action="start" />;
+	next();
+}
+
+export async function desktopLoginFinalize( context, next ) {
+	const { hash } = context;
+	context.primary = (
+		<DesktopLogin action="finalize" error={ hash?.error } accessToken={ hash?.access_token } />
+	);
+	return next();
+}
```

Comment:
The two new route handlers don't need to be async functions: there is no `await` inside them. They pass control by calling the `next()` function, and the return value (promise) from the handler is not used anywhere.

`return next()` can be replaced with `next()`: the function will return anyway on the next line.

Reply:
Good points, fixed.